fast64_internal/f3d/f3d_writer.py

```diff
@@ -1313,12 +1313,26 @@ def saveOrGetF3DMaterial(material, fModel, obj, drawLayer, convertTextureData):
         + (("_layer" + str(drawLayer)) if f3dMat.rdp_settings.set_rendermode and drawLayer is not None else "")
         + (("_area" + str(areaIndex)) if f3dMat.set_fog and f3dMat.use_global_fog and areaKey is not None else "")
     )
-    fMaterial = fModel.addMaterial(materialName)
-    fMaterial.mat_only_DL.commands.append(DPPipeSync())
-    fMaterial.revert.commands.append(DPPipeSync())
 
     if not material.is_f3d:
         raise PluginError("Material named " + material.name + " is not an F3D material.")
+    fMaterial = fModel.addMaterial(materialName)
+    useDict = all_combiner_uses(f3dMat)
+
+    defaults = bpy.context.scene.world.rdp_defaults
+    if fModel.f3d.F3DEX_GBI_2:
+        saveGeoModeDefinitionF3DEX2(fMaterial, f3dMat.rdp_settings, defaults, fModel.matWriteMethod)
+    else:
+        saveGeoModeDefinition(fMaterial, f3dMat.rdp_settings, defaults, fModel.matWriteMethod)
+
+    # Checking for f3dMat.rdp_settings.g_lighting here will prevent accidental exports,
+    # There may be some edge case where this isn't desired.
+    if useDict["Shade"] and f3dMat.set_lights and f3dMat.rdp_settings.g_lighting:
```

Comment:
extreme nitpick but id change this to
```suggestion
    if useDict["Shade"] and f3dMat.rdp_settings.g_lighting and f3dMat.set_lights:
```

Reply:
Sure 👍